test_conformance/subgroups/test_workgroup_non_uniform_vote.cpp

```diff
@@ -0,0 +1,504 @@
+//
+// Copyright (c) 2021 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "procs.h"
+#include "subhelpers.h"
+#include "harness/typeWrappers.h"
+
+// Global/local work group sizes
+// Adjust these individually below if desired/needed
+#define GWS 2000
+#define LWS 200
+
+#define GWS_NON_UNIFORM 170
+#define LWS_NON_UNIFORM 64
+namespace {
+template <typename T>
+void calculate(const T *mx, NonUniformVoteOp operation, int subgroup_size,
+               bool *compare_result)
+{
+    compare_result[0] = 1;
+    compare_result[1] = 1;
+    switch (operation)
+    {
+        case NonUniformVoteOp::all_equal:
+            for (int i = 0; i < subgroup_size; ++i)
+            {
+                if (i < NR_OF_ACTIVE_WORK_ITEMS)
+                { // return non zero if all the same
+                    compare_result[0] &= compare_ordered(mx[i], mx[0]);
+                }
+                else
+                {
+                    compare_result[1] &=
+                        compare_ordered(mx[i], mx[NR_OF_ACTIVE_WORK_ITEMS]);
+                }
+            }
+            break;
+        default: log_error("Unknown operation request"); break;
+    }
+}
+template <>
+void calculate(const int *mx, NonUniformVoteOp operation, int subgroup_size,
```

Comment:
Why do we need a template specialisation here? Can't all the code be in the above function? It seems this copy/paste is not necessary.